File: solutions/javascript/nucleotide-count/1/nucleotide-count.js
//
// This is only a SKELETON file for the 'Nucleotide Count' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export function countNucleotides(strand) {
  var a = 0;
  var c = 0
  var g = 0;
  var t = 0;
  for (var ch of strand) {
    switch(ch) {
      case 'A': a++; break;
      case 'C': c++; break;
      case 'G': g++; break;
      case 'T': t++; break;
      default: throw new Error('Invalid nucleotide in strand');
    }
  }
  return `${a} ${c} ${g} ${t}`;
}

File: solutions/julia/nucleotide-count/1/nucleotide-count.jl
"""
    count_nucleotides(strand)

The count of each nucleotide within `strand` as a dictionary.

Invalid strands raise a `DomainError`.

"""
function count_nucleotides(strand)
    nc = Dict('A' => 0, 'C' => 0, 'G' => 0, 'T' => 0)
    for d in strand
        if get(nc, d, "Error") == "Error"
            throw(DomainError(d))
        else
            nc[d] = nc[d]+1
        end
    end
    return nc
end
